pkg/osinfo/container_scanner.go

```diff
@@ -108,6 +114,40 @@ func (ct *ContainerScanner) ReadDebianPackages(layers []string) (layer int, pk *
 	return layer, pk, err
 }
 
+// ReadApkPackages reads the last known changed copy of the apk database
+func (ct *ContainerScanner) ReadApkPackages(layers []string) (layer int, pk *[]PackageDBEntry, err error) {
+	apkDatabase := ""
+	loss := LayerScanner{}
+	for i, lp := range layers {
+		tmpDB, err := os.CreateTemp("", "apkdb-")
+		if err != nil {
+			return 0, pk, fmt.Errorf("opening temporary apkdb file: %w", err)
+		}
+		tmpDBPath := tmpDB.Name()
+		defer os.Remove(tmpDBPath)
+		if err := loss.extractFileFromTar(lp, apkDBPath, tmpDBPath); err != nil {
+			if _, ok := err.(ErrFileNotFoundInTar); ok {
+				continue
+			}
+			return 0, pk, fmt.Errorf("extracting apk database: %w", err)
+		}
+		logrus.Debugf("Layer %d has a newer version of apk database", i)
+		apkDatabase = tmpDBPath
+		layer = i
+	}
+
+	if apkDatabase == "" {
+		logrus.Info("apk database data is empty")
+		return layer, nil, nil
+	}
+
+	pk, err = ct.parseApkDB(apkDatabase)
+	if err != nil {
+		return layer, nil, fmt.Errorf("parsing apk database: %w", err)
+	}
+	return layer, pk, err
```

Comment:
I'd make this explicit:
```suggestion
	return layer, pk, nil
```